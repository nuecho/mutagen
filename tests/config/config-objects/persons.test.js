const checkKey = (cfgObject) => {return cfgObject.employeeId === "cfgObjectTest"};

cfgObjectTests("persons", "config-objects/persons-config.json", checkKey, 1);
