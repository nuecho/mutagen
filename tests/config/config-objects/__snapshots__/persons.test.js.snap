// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen persons config import-export] should create the persons: the persons have been created 1`] = `
Object {
  "stderr": "",
  "stdout": "+ person => Environment/cfgObjectTest
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen persons config import-export] should throw exception when new persons to import miss mandatory properties persons: exception has been thrown 1`] = `
Object {
  "stderr": "Cannot import configuration: some configuration objects' mandatory properties for creation are not set.
  Properties [userName] not set in person [Environment/emptyCfgObjectTest].
",
  "stdout": "",
}
`;

exports[`[mutagen persons config import-export] should update the modified persons: the modified persons have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ person => Environment/cfgObjectTest
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen persons config import-export] should update the same persons: the persons have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ person => Environment/cfgObjectTest
Completed. 1 object(s) imported.
",
}
`;
